### ISSUE TITLE
Error in spirv_hip_fp16.h with -D__HIP_NO_HALF_CONVERSIONS__=1

### ISSUE THREAD

**Issue by linehill** [CHIP-SPV/chip-spv#362]
- [x] merge #401 
- [ ] merge #396 
- [ ] check if failing
- [x] remove `TestHipccHalfConversions` and `TestHipccHalfOperators` tests from exclusion list

The following snippet:

```
#include <hip/hip_runtime.h>
#include <hip/hip_fp16.h>
```
Compiles in ROCm/hipcc but not in CHIP-SPV/hipcc with -D__HIP_NO_HALF_CONVERSIONS__=1 on the latest main (9a684ed3).

```
$ ./tandem-hipcc -c -D__HIP_NO_HALF_CONVERSIONS__=1 hip-fp16-include.hip 
vvv Running /opt/rocm/bin/hipcc -c -D__HIP_NO_HALF_CONVERSIONS__=1 hip-fp16-include.hip
^^^ DONE (exit code: 0)

vvv Running <CHIP-SPV-install>/bin/hipcc -c -D__HIP_NO_HALF_CONVERSIONS__=1 hip-fp16-include.hip
Hip Clang Compiler not found
In file included from hip-fp16-include.hip:2:
In file included from /mnt/md1/linehill/ws-chip-spv-2/install/include/hip/hip_fp16.h:41:
/mnt/md1/linehill/ws-chip-spv-2/install/include/hip/spirv_hip_fp16.h:1586:21: error: no matching conversion for C-style cast from 'float' to '__half'
                    __llvm_amdgcn_rcp_f16(static_cast<__half_raw>(x).data)};
                    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
...
^^^ DONE (exit code: 1)
```
The compilation will probably fail on the the AMDGCN intrinsic too in a later compilation phase.

**Comment by linehill** [comment]
Same error with `-D__HIP_NO_HALF_OPERATORS__=1` too.

**Comment by pvelesko** [comment]
Looks like this conversion is not valid:
```
#define __llvm_amdgcn_rcp_f16(__X) ((__half)1.0f / (__half)__X)
#define __llvm_amdgcn_rcp_2f16(__X) ((__half2)1.0f / (__half2)__X)
```

```
/home/pvelesko/CHIP-SPV/hipccFixes/build/include/hip/spirv_hip_fp16.h:1586:21: error: no matching constructor for initialization of '__half'
17:                     __llvm_amdgcn_rcp_f16(static_cast<__half_raw>(x).data)};
17:                     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
17: /home/pvelesko/CHIP-SPV/hipccFixes/build/include/hip/spirv_hip_fp16.h:72:37: note: expanded from macro '__llvm_amdgcn_rcp_f16'
17: #define __llvm_amdgcn_rcp_f16(__X) (__half{1} / (__half)__X)
17:                                     ^     ~~~
17: /home/pvelesko/CHIP-SPV/hipccFixes/build/include/hip/spirv_hip_fp16.h:159:13: note: candidate constructor not viable: no known conversion from 'int' to 'const __half_raw' for 1st argument
17:             __half(const __half_raw& x) : data{x.data} {}
17:             ^
17: /home/pvelesko/CHIP-SPV/hipccFixes/build/include/hip/spirv_hip_fp16.h:170:13: note: candidate constructor not viable: no known conversion from 'int' to 'const __half' for 1st argument
17:             __half(const __half&) = default;
17:             ^
17: /home/pvelesko/CHIP-SPV/hipccFixes/build/include/hip/spirv_hip_fp16.h:172:13: note: candidate constructor not viable: no known conversion from 'int' to '__half' for 1st argument
17:             __half(__half&&) = default;
17:             ^
17: /home/pvelesko/CHIP-SPV/hipccFixes/build/include/hip/spirv_hip_fp16.h:157:13: note: candidate constructor not viable: requires 0 arguments, but 1 was provided
17:             __half() = default;
```

to be addressed in #396 unless @pjaaskel you have a quick fix?
